typescript/sdk/src/deploy/verify/ContractVerifier.ts

```diff
@@ -7,167 +7,109 @@ import { sleep, strip0x } from '@hyperlane-xyz/utils';
 import { ExplorerFamily } from '../../metadata/chainMetadataTypes';
 import { MultiProvider } from '../../providers/MultiProvider';
 import { ChainMap, ChainName } from '../../types';
-import { MultiGeneric } from '../../utils/MultiGeneric';
 
 import {
   CompilerOptions,
   ContractVerificationInput,
-  VerificationInput,
+  ExplorerApiActions,
+  ExplorerApiErrors,
+  ExplorerLicenseType,
+  FormOptions,
 } from './types';
 
-enum ExplorerApiActions {
-  GETSOURCECODE = 'getsourcecode',
-  VERIFY_IMPLEMENTATION = 'verifysourcecode',
-  MARK_PROXY = 'verifyproxycontract',
-  CHECK_STATUS = 'checkverifystatus',
-  CHECK_PROXY_STATUS = 'checkproxyverification',
-}
-
-enum ExplorerApiErrors {
-  ALREADY_VERIFIED = 'Contract source code already verified',
-  ALREADY_VERIFIED_ALT = 'Already Verified',
-  VERIFICATION_PENDING = 'Pending in queue',
-  PROXY_FAILED = 'A corresponding implementation contract was unfortunately not detected for the proxy address.',
-  BYTECODE_MISMATCH = 'Fail - Unable to verify. Compiled contract deployment bytecode does NOT match the transaction deployment bytecode.',
-}
-
-export class ContractVerifier extends MultiGeneric<VerificationInput> {
-  protected logger: Debugger;
+export class ContractVerifier {
+  private logger = debug(`hyperlane:ContractVerifier`);
+  private compilerOptions: CompilerOptions;
 
   constructor(
-    verificationInputs: ChainMap<VerificationInput>,
-    protected readonly multiProvider: MultiProvider,
-    protected readonly apiKeys: ChainMap<string>,
-    protected readonly source: string, // solidity standard input json
-    protected readonly compilerOptions: CompilerOptions,
+    private multiProvider: MultiProvider,
+    private readonly apiKeys: ChainMap<string>,
+    private source: string, // solidity standard input json
```

Comment:
jmrossy
> Why make these private? I don't think we need member data protection but it's conceivable that we'd want to extend this class one day

Reply:
I kind of went on a tear trying to lock down the class interface, but you're right that we should keep this protected readonly